lib/child_process.js

```diff
@@ -315,6 +315,9 @@ function _convertCustomFds(options) {
 function normalizeSpawnArguments(file /*, args, options*/) {
   var args, options;
 
+  if (typeof file !== 'string' || file.length === 0)
+    throw new TypeError('"file" argument must be a string');
```

Comment:
Nit: If the input is an empty string, the error message might not be exactly appropriate.